### ISSUE TITLE
Bad syntax in function signature not reported

### ISSUE THREAD

**Issue by marcelka** [eslint/eslint#4067]
Eslint (v1.6.0) does not report invalid syntax for the following piece of code:
```
function someFn({someArg: {}}) { // I used ':' instead of '='
  // much code
}
someFn()
```

eslintrc used:
```{"parser": "babel-eslint"}```

However, afaik not-reporting-the-error is independent of what rules I enable/disable in the eslintrc file.

Since the input file is not a valid js file (I guess it should not be possible to create valid AST from this), I'd expect some 'invalid code' error.

**Comment by eslintbot** [comment]
Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by mysticatea** [comment]
Thank you for reporting.

This is duplicate of #3658.
It's valid ECMAScript code :)
The empty object pattern is allowed in the spec: http://www.ecma-international.org/ecma-262/6.0/#sec-destructuring-assignment

However, we have a rule to report that pattern in master (#3668), so please wait for the next release!

**Comment by marcelka** [comment]
Oh, thanks for pointing that out, I haven't noticed that it is in fact valid code :) I am looking forward to the rule!
